app/helpers/layout_helper.rb

```diff
@@ -44,6 +44,19 @@ def collapsing_header(title, target, collapsed = '')
     end
   end
 
+  def resource_title(resource_name)
```

Comment:
Hmm... how do we do this for the navigation?

Reply:
it's hard coded strings for every link - e.g. https://github.com/theforeman/foreman/blob/develop/app/registries/menu/loader.rb#L67

Reply:
Is there any chance of accessing this data from here? Seems to me like that'd be an easy solution that would cover all the resources, including those from plugins.

Reply:
Theoretically, this can be done, but it would require implementing some sort of nested searching in the `Menu::Manager`, something of the sort of (this is ugly code that i hacked together in 3 minutes, it can probably be written better):
```ruby
def find_caption(resource)
  @items[:top_menu].children.map do |submenu| 
    submenu.children.find{|item| item.name == resource}&.caption
  end.compact.first
end
```